assets/js/index.js

```diff
@@ -0,0 +1,157 @@
+// Assignment Code
+const generateBtn = document.querySelector("#generate");
+
+//function - get the password length from the user
+const getPasswordLength = () => {
+  //prompt user to give a userInput
+  //declare variable to receive user input - const -> not modified further down
+  const userInput = prompt(
+    "How long would you like your password to be? \n\nPlease enter a number between 8 and 128.\n",
+    ""
+  );
+  // check if userInput is a number
+  const passwordLength = parseInt(userInput);
+
+  // converts userInput into a number (or return NaN if not a number)
+  if (!isNaN(passwordLength) && passwordLength >= 8 && passwordLength <= 128) {
+    //if type and range are both true, the number is returned for storage in variable
+    return passwordLength;
+  }
+  //if type or range are not true, it sends an alert box with message
+  else {
+    alert(
+      "You must enter a number between 8 and 128. \n\nClick OK to start again."
+    );
+    //click ok on alert box to reload the page and start again
+    window.location.reload(true);
+    document.getElementById("password").reset();
```

Comment:
This line also throws a JS error in the console. Removing this feature should prevent that bug

`TypeError: document.getElementById(...).reset is not a function`

Reply:
hi, when I remove that line, it removes the console error, but it also doesn't refresh the page, it actually just continues showing the prompts for the next questions after I close the password length alert. How can I fix that and actually find a way to stop after the password length when not good?

Reply:
never mind, I resolved it! you can have a look and let me know what you think if you like :)